src/main/java/org/joshsim/lang/interpret/visitor/JoshConfigDiscoveryVisitor.java

```diff
@@ -0,0 +1,136 @@
+package org.joshsim.lang.interpret.visitor;
+
+import java.util.HashSet;
+import java.util.Set;
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.joshsim.engine.config.DiscoveredConfigVar;
+import org.joshsim.lang.antlr.JoshLangBaseVisitor;
+import org.joshsim.lang.antlr.JoshLangParser;
+
+/**
+ * Visitor to discover configuration variables used in Josh scripts.
+ *
+ * <p>This visitor traverses the AST and collects all configuration variable
+ * references found in config expressions. For example, "config example.testVar"
+ * would add "example.testVar" to the discovered set, while "config example.testVar else 5 m"
+ * would add "example.testVar(5 m)" to show the default value.</p>
+ */
+public class JoshConfigDiscoveryVisitor extends JoshLangBaseVisitor<Set<DiscoveredConfigVar>> {
```

Comment:
Across this whole class, there are places where we are doing inspection or pseudo-inspection. We should do a web search and look at the other visitors in this project. Ideally, the use of the visitor pattern could handle this complexity without inspecting class names or checking if attributes are present. However, we should evaluate if the alternative if viable.